napari/layers/labels/labels.py

```diff
@@ -449,8 +449,16 @@ def fill(self, coord, new_label):
             Position of mouse cursor in image coordinates.
         new_label : int
             Value of the new label to be filled in.
+        refresh : bool
+            Whether to refresh view slice or not. Set to False to batch paint
+            calls.
         """
-        self._save_history()
+        if refresh is True:
+            self._save_history()
+
+        old_label = self.get_value()
```

Comment:
The `get_value` call needs to come after the `self.coordinates = slice_coord` update otherwise it will use the old `coordinates`, we might want to refactor and improve this at some point as it is not intuitive/ could lead to some surprising bugs, but for now can you move the call down lower. The slicing code should be fast, so it's ok. The slow line is what used to be 471.

Reply:
yep, the advanced test caught it! corrected it since